trieste/acquisition/function/function.py

```diff
@@ -762,3 +763,74 @@ def acquisition(x: TensorType) -> TensorType:
         return tf.exp(tf.linalg.logdet(covariance + jitter))
 
     return acquisition
+
+
+class BALD(SingleModelAcquisitionBuilder):
+    """
+    Builder for the *Bayesian Active Learning By Disagreement* acquisition function defined in
+    :cite:`houlsby2011bayesian`. The acquisition function computes the information gains of the predictive entropy.
+
+    This acquisition function is intended to use for Binary Gaussian Process Classification model with Bernoulli likelihood.
+    """
+
+    def __init__(self, jitter: float = DEFAULTS.JITTER) -> None:
+        """
+        :param jitter: The size of the jitter to avoid numerical problem caused by the variance value.
+        """
+        self._jitter = jitter
+
+    def __repr__(self) -> str:
+        """"""
+        return f"BALD(jitter={self._jitter!r})"
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model.
+        :param dataset: Unused.
+
+        :return: The determinant of the predictive function.
+        """
+
+        return bald(model, self._jitter)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: Unused.
+        """
+        return function  # no need to update anything
+
+
+class bald(AcquisitionFunctionClass):
+    def __init__(self, model: ProbabilisticModel, jitter: float):
+        """
+
+        :param model: The model of the objective function.
+        :param jitter: The size of the jitter to avoid numerical problem caused by the variance value.
+        :return: The Bayesian Active Learning By Disagreement acquisition function.
+        """
+        self._model = model
+        self._jitter = jitter
+
+    # @tf.function
```

Comment:
there is an issue with VGP that is the root cause of retracing (see #406), it might not be due to the code here, @uri-granta is working on fixing this